src/praisonai-bot/praisonai_bot/gateway/server.py

```
@@ -4466,6 +4497,157 @@ async def _cleanup():
+    def _load_persisted_session_data(self, session_id: str) -> Optional[Dict[str, Any]]:
+        """Return the latest persisted ``session_data`` snapshot for a session.
+
+        Mirrors the snapshot lookup in :meth:`create_session`: the newest
+        ``system`` message carrying a ``session_data`` metadata blob wins.
+        Returns ``None`` when unavailable.
+        """
+        if not self._session_store:
+            return None
+        try:
+            session_obj = self._session_store.get_session(session_id)
+        except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)